Add reddit search provider

mig5 opened the pull request:
A little Reddit search provider which makes use of API as described at https://www.reddit.com/dev/api#GET_search

ahoernecke commented:
Hi @mig5, great plugin! There seems to be a bug for cases where a subreddit is specified but the number of max results is not. If you could cast @options[:results] to string (@options[:results].to_s) where you're generating the url, that should fix the issue.

Side note, I just created a develop branch. If you have future pull requests and submit to that branch I should be able to fix simple issues like this :-)

Thanks again!

ahoernecke commented on lib/search_providers/reddit.rb:
```diff
@@ -0,0 +1,59 @@
+    if(@options[:subreddit].blank?)
+      url = URI.escape('https://www.reddit.com/search.json?q=' + @query)
+    else
+      url = URI.escape('https://www.reddit.com/r/' + @options[:subreddit] + '/search.json?q=' + @query + '&limit=' + @options[:results])
```
Update to: 
```
url = URI.escape('https://www.reddit.com/r/' + @options[:subreddit] + '/search.json?q=' + @query + '&limit=' + @options[:results].to_s)
```

mig5 commented:
Thanks for the review @ahoernecke, updated. I also realised I was not sending the Max Results to the search URL for the non-subreddit search too, so that's also fixed. I assume the same .to_s is needed there too (sorry, a bit new to Ruby)

ahoernecke commented:
Looks great, thanks again!
